commit 0e467869f4d427a8c42e1c2ff6a0bb215f288b09
Author: James Turnbull <james@lovedthanlost.net>
Date:   Sat Feb 28 03:05:23 2009 +1100

     Fixed #1963 - Failing to read /proc/mounts for selinux kills file downloads

diff --git a/lib/puppet/util/selinux.rb b/lib/puppet/util/selinux.rb
index 70f2445..cd3b2ac 100644
--- a/lib/puppet/util/selinux.rb
+++ b/lib/puppet/util/selinux.rb
@@ -153,7 +153,9 @@ module Puppet::Util::SELinux
     # Internal helper function to read and parse /proc/mounts
     def read_mounts
         begin
-            mounts = File.read("/proc/mounts")
+            mountfh = File.open("/proc/mounts", NONBLOCK)
+            mounts = mountfh.read
+            mountfh.close
         rescue
             return nil
         end
